presto-main/src/main/java/com/facebook/presto/execution/WarningSink.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.spi.WarningCode;
+
+import java.util.List;
+
+public interface WarningSink
```

Comment:
This interface is interesting. It has `sink` its name, but it is also readable. I think we can rework this API around `Consumer<PrestoWarning>` and `Supplier<List<PrestoWarning>>` and also rename it:

```
public interface WarningCollector
    extends Consumer<PrestoWarning>, Supplier<List<PrestoWarning>>
{
}
```